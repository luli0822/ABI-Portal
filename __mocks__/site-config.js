module.exports = {
	assetPrefix: '/some-asset-prefix',
};
