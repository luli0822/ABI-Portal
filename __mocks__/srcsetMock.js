module.exports = {
	sources: {
		'992w': 'mock-992w-img-src',
	},
	srcSet: 'mock-src-set',
};
