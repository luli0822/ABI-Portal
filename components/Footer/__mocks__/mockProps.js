import { pageLinkMock, externalLinkMock } from '../../../.storybook/SiteData';

export const navigationEntries = [
	{
		type: 'link',
		link: pageLinkMock,
		text: 'Link 1',
		accessibleText: undefined,
	},
	{
		type: 'link',
		link: externalLinkMock,
		text: 'Link 2',
		accessibleText: undefined,
	},
];

export const defaultProps = {
	navigationEntries,
};
