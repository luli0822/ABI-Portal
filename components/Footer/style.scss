@import "../../scss/a11y-mixins";
@import "../../scss/button-link-mixins";
@import "../../scss/layout-mixins";
@import "../../scss/text-mixins";

$rootClass: "footer";

.#{$rootClass} {
  background: black;
  flex: 0 0 auto;
  // This is to position the footer to the bottom of the page
  display: flex;
  flex-direction: column;
  justify-content: flex-end;

  &__container {
    @include container();

    padding: 40px 0;
  }

  &__nav-list {
    display: flex;
    justify-content: center;

    @include media-breakpoint-up(md) {
      justify-content: flex-end;
    }
  }

  &__nav-list-item {
    padding-right: 30px;

    &:last-child {
      padding-right: 0;
    }

    @include media-breakpoint-up(md) {
      &:last-child {
        padding-right: 30px;
      }
    }
  }

  &__nav-link {
    text-decoration: none;
    color: $white;
  }
}
