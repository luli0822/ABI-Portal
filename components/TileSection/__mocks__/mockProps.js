import { imagePNGMock } from '../../../.storybook/SiteData';
// eslint-disable-next-line import/prefer-default-export

export const primaryMedia = {
	primaryMedia: imagePNGMock,
	primaryLinkAccessibleText: `Image Accessible Text`,
};

export const defaultBlocks = [
	{
		contentType: 'Complex Block',
		id: 'tileBlock1',
		title: 'Tile 1',
		body: {
			// NOTE: For richText body, the CustomDoc component will render the content from values:[] field to eliminate the complex mapping for body.content fields to read the content.
			// For paragraph - {type: 'paragraph' , value: ''}, for list items {type: 'list', value: ['', '']}
			values: [
				{
					type: 'paragraph',
					value: 'Tile 1 Body',
				},
			],
			nodeType: 'document',
			data: {},
			content: [
				{
					nodeType: 'paragraph',
					data: {},
					content: [
						{
							nodeType: 'text',
							value: 'Tile 1 Body',
							data: {},
							marks: [],
						},
					],
				},
			],
		},
		...primaryMedia,
	},
	{
		contentType: 'Complex Block',
		id: 'tileBlock2',
		title: 'Tile 2',
		body: {
			// NOTE: For richText body, the CustomDoc component will render the content from values:[] field to eliminate the complex mapping for body.content fields to read the content.
			// For paragraph - {type: 'paragraph' , value: ''}, for list items {type: 'list', value: ['', '']}
			values: [
				{
					type: 'paragraph',
					value: 'Tile 2 Body',
				},
			],
			nodeType: 'document',
			data: {},
			content: [
				{
					nodeType: 'paragraph',
					data: {},
					content: [
						{
							nodeType: 'text',
							value: 'Tile 2 Body',
							data: {},
							marks: [],
						},
					],
				},
			],
		},
		...primaryMedia,
	},
];

export const defaultContent = {
	contentType: 'Block Group',
	title: 'What can we build for you?',
	body: {
		values: [
			{
				type: 'paragraph',
				value: 'From microsites to interactive quizzes - we know digital experience.',
			},
		],
		nodeType: 'document',
		data: {},
		content: [
			{
				nodeType: 'paragraph',
				data: {},
				content: [
					{
						nodeType: 'text',
						value: 'From microsites to interactive quizzes - we know digital experience.',
						data: {},
						marks: [],
					},
				],
			},
		],
	},
};

// Default props without links
export const defaultProps = {
	content: {
		...defaultContent,
		blocks: defaultBlocks,
	},
};
