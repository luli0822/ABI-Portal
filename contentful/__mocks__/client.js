import fetchSiteData from './fetchSiteData.json';
// Refresh mock siteData will disclude about-us
module.exports = {
	getClient: () => {},
	fetchSiteData: () => fetchSiteData,
};
