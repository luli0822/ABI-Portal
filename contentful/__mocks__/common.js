// A method to build Link data structure
export const linkDataMock = (type, linkId = 'link-id-1') => ({
	sys: {
		id: linkId,
		contentType: {
			sys: {
				type: 'Link',
				linkType: 'ContentType',
				id: `${type}`,
			},
		},
	},
	fields: {
		title: {
			'en-CA': 'Link Text',
		},
		accessibleLabel: {
			'en-CA': 'Link accessible Text',
		},
		url: type === 'externalLink' ? { 'en-CA': 'https://loblawdigital.co' } : undefined,
		page:
			type === 'pageLink'
				? {
						'en-CA': {
							sys: {
								type: 'Link',
								linkType: 'Entry',
								id: 'site-home-page',
							},
						},
				  }
				: undefined,
	},
});

// Use this for pageLink structure mock
export const pageLinkMock = linkDataMock('pageLink');

// Use this externalLink structure mock
export const externalLinkMock = linkDataMock('externalLink');

// A method to build Media/image data structure
export const imageDataMock = type => ({
	sys: {
		type: 'Asset',
		id: type === 'image/png' ? 'asset-id-1' : 'asset-id-2',
	},
	fields: {
		title: {
			'en-CA': 'Image Title Text',
		},
		description: {
			'en-CA': 'Image description Text',
		},
		file: {
			'en-CA': {
				url: 'image-file-url.png',
				details: {
					size: 150795,
					image: {
						width: 300,
						height: 300,
					},
				},
				fileName: 'png-example.png',
				contentType: type,
			},
		},
	},
});
