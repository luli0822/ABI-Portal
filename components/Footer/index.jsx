import React from 'react';
import PropTypes from 'prop-types';
import ContentfulLink from '../../contentful/ContentfulLink';
import './style.scss';

const rootClass = 'footer';

function Footer(props) {
	const { navigationEntries } = props;

	return (
		<footer id="footer" className={rootClass}>
			<div className={`${rootClass}__container`}>
				<div className={`${rootClass}__nav-links-container`}>
					<ul className={`${rootClass}__nav-list`}>
						{navigationEntries.map(entry => (
							<li className={`${rootClass}__nav-list-item`} key={entry.link.sys.id}>
								<ContentfulLink linkEntry={entry.link} className={`${rootClass}__nav-link`}>
									{entry.text}
								</ContentfulLink>
							</li>
						))}
					</ul>
				</div>
			</div>
		</footer>
	);
}

Footer.propTypes = {
	navigationEntries: PropTypes.arrayOf(
		PropTypes.shape({
			fields: PropTypes.shape({
				page: PropTypes.object.isRequired,
				title: PropTypes.object.isRequired,
			}),
		})
	),
};

Footer.defaultProps = {
	navigationEntries: [],
};

export default Footer;
