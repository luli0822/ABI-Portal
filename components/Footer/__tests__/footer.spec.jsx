import React from 'react';
import { shallow } from 'enzyme';
import Footer from '../index';
import { defaultProps } from '../__mocks__/mockProps';

describe('<Footer />', () => {
	it('renders correctly', () => {
		// eslint-disable-next-line react/jsx-props-no-spreading
		expect(shallow(<Footer {...defaultProps} />).debug()).toMatchSnapshot();
	});
});
