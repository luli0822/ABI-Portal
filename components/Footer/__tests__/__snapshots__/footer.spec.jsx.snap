// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders correctly 1`] = `
"<footer id=\\"footer\\" className=\\"footer\\">
  <div className=\\"footer__container\\">
    <div className=\\"footer__nav-links-container\\">
      <ul className=\\"footer__nav-list\\">
        <li className=\\"footer__nav-list-item\\">
          <ContentfulLink linkEntry={{...}} className=\\"footer__nav-link\\" targetLang={[undefined]}>
            Link 1
          </ContentfulLink>
        </li>
        <li className=\\"footer__nav-list-item\\">
          <ContentfulLink linkEntry={{...}} className=\\"footer__nav-link\\" targetLang={[undefined]}>
            Link 2
          </ContentfulLink>
        </li>
      </ul>
    </div>
  </div>
</footer>"
`;
